pkg/arvo/sys/hoon.hoon

```diff
@@ -4181,6 +4181,21 @@
 ::
 ++  crip  |=(a=tape `@t`(rap 3 a))                      ::  tape to cord
 ::
+++  edit                                                :: cord replace
+  |=  [src=@t get=@t set=@t all=?]
+  ^-  @t
+  ?:  =(get set)  src
+  =+  cut=(lent (trip get))
+  |-
+  =+  i=(find (trip get) (trip src))
+  ?~  i  src
+  ?.  all
+    (crip (into (oust [+.i cut] (trip src)) +.i set))
+  ?~  (find (trip get) (trip set))
+    $(src (crip (into (oust [+.i cut] (trip src)) +.i set)))
+  ~&  >>  'infinite loop: set must not contain get'
+  src
+::
```

Comment:
- For cords, using the binary arithmetic operations is always going to be significantly more efficient
- You likely want to crash if `get` is empty
- Since we expect `get` and `set` to not be identical, I think it might actually be better to just never check if they're identical. A user who does this almost certainly has an upstream error that he should correct himself, rather than us correcting it for him silently in `edit` at the cost of a direct comparison for every single usage at every call site.

```
++  edit                                                :: cord replace
  |=  [src=@t get=@t set=@t all=?]
  ^-  @t
  =/  metg  (met 3 get)
  =/  mets  (met 3 set)
  ?~  metg  !!
  =/  len   (met 3 src)
  =/  pos  0
  |-
  ?:  =(pos len)
    src
  ?.  =((cut 3 [pos metg] src) get)
    $(pos +(pos))
  =.  src
    (rap 3 ~[(cut 3 [0 pos] src) set (rsh [3 (add pos metg)] src)])
  ?.  all
    src
  $(pos (add pos mets), len (add mets (sub len metg)))
```

Reply:
Ah, bit arithmetic. Enlightening, thank you.

I've omitted ++swap for now, and applied the improvements to ++edit.